src/BoundaryConditions/fill_halo_regions.jl

```diff
@@ -21,6 +21,8 @@ function fill_halo_regions!(fields::Union{Tuple, NamedTuple}, arch, args...)
     return nothing
 end
 
+fill_halo_regions!(c::OffsetArray, ::Nothing, args...; kwargs...) = nothing
```

Comment:
Just adding a comment to provide context. Please correct if wrong!
```suggestion
# `FunctionField`s and `ZeroField`s have no boundary conditions.
fill_halo_regions!(c::OffsetArray, ::Nothing, args...; kwargs...) = nothing
```

Reply:
That's a good comment, I will just future proof it by saying "Some fields have `nothing` boundary conditions, such as `FunctionField` and `ZeroField`." (eg we might add more fields in the future with `nothing` boundary conditions).